test_conformance/images/kernel_read_write/test_common.cpp

```diff
@@ -141,17 +175,52 @@ void InitFloatCoordsCommon(image_descriptor *imageInfo,
                     {
                         xOffsets[i] =
                             (float)((double)xOffsets[i] / (double)width_lod);
-                        yOffsets[i] =
-                            (float)((double)yOffsets[i] / (double)height_lod);
-                        zOffsets[i] =
-                            (float)((double)zOffsets[i] / (double)depth_lod);
+                        if (imageInfo->type != CL_MEM_OBJECT_IMAGE1D_ARRAY)
+                        {
+                            yOffsets[i] = (float)((double)yOffsets[i]
+                                                  / (double)height_lod);
+                        }
+                        if (imageInfo->type != CL_MEM_OBJECT_IMAGE2D_ARRAY)
+                        {
+                            zOffsets[i] = (float)((double)zOffsets[i]
+                                                  / (double)depth_lod);
+                        }
                     }
                 }
             }
         }
     }
 }
 
+cl_mem create_image_of_type(cl_context context, cl_mem_flags mem_flags,
+                            image_descriptor *imageInfo, size_t row_pitch,
+                            size_t slice_pitch, void *host_ptr, cl_int *error)
+{
+    cl_mem image;
+    switch (imageInfo->type)
+    {
+        case CL_MEM_OBJECT_IMAGE3D:
+            image = create_image_3d(context, mem_flags, imageInfo->format,
+                                    imageInfo->width, imageInfo->height,
+                                    imageInfo->depth, row_pitch, slice_pitch,
+                                    host_ptr, error);
+            break;
+    }
+    return image;
+}
+
+static size_t get_image_num_pixels(image_descriptor *imageInfo, size_t width,
+                                   size_t height, size_t depth,
+                                   size_t array_size)
+{
+    size_t image_size;
+    switch (imageInfo->type)
+    {
+        case CL_MEM_OBJECT_IMAGE3D: image_size = width * height * depth; break;
```

Comment:
We should add a default case to handle unsupported image types (especially since only 3D is supported now. Would be nice to add a comment noting that the implementation is incomplete.

Reply:
Done. 
Hopefully this is as expected, but it will be removed in a follow up pull request anyway.